src/NuGet.Clients/PackageManagement.UI/Common/PackageCollection.cs

```diff
@@ -86,6 +87,25 @@ public static PackageIdentity[] GetEarliest(this IEnumerable<PackageIdentity> pa
                 .Select(g => new PackageIdentity(g.Key, g.MinOrDefault()))
                 .ToArray();
         }
+
+        /// <summary>
+        /// True if any package reference is AutoReferenced=true.
+        /// </summary>
+        public static bool IsAutoReferenced(this IEnumerable<PackageCollectionItem> packages, string id)
+        {
+            return packages.Where(e => StringComparer.OrdinalIgnoreCase.Equals(id, e.Id))
+                .Any(e => e.IsAutoReferenced());
+        }
+    }
+
+    internal static class PackageCollectionItemExtensions
+    {
+        public static bool IsAutoReferenced(this PackageCollectionItem package)
```

Comment:
It could be either, I think it fits better here with the rest of the helper methods for these classes.